### PR TITLE
flag autoreset on in the service

### PR DESCRIPTION
**Pull request by buoyad**
@keybase/y2ksquad

### CODE REVIEW

**Comment by buoyad** [comment]
Interested if there's a way in `features.go` to hardcode it on rather than changing it at access sites like this PR currently does.

File: go/engine/passphrase_recover.go [hunk 1]
```diff
@@ -65,7 +65,8 @@ func (e *PassphraseRecover) Run(mctx libkb.MetaContext) (err error) {
 	}
 
 	// If the reset pipeline is not enabled, we'll want this to act exactly the same way as before
-	if !mctx.G().Env.GetFeatureFlags().HasFeature(libkb.EnvironmentFeatureAutoresetPipeline) {
+	// Autoreset hardcoded on for now.
+	if !true {
```
Comment (heronhaye) [inline go/engine/passphrase_recover.go]: long term we should get rid of these branches and make it appear as if the feature flag never existed

**Comment by zapu** [comment]
kk let me take over for a bit here. looking at the PGP login tests

**Comment by zapu** [comment]
I fixed some of the stuff but it may be breaking others because it changes return values of engines in some flows.

I'd typically want to avoid no-ops in engines returning `nil`, unless it's some kind of idempotent operation. For example, we had `recover passphrase` here that had "you can't recover because you are logged in - doing nothing". I made this an error rather than no-error.

**Comment by zapu** [comment]
It's only failing on `TestBackgroundPurge` now which is a flake.

@keybase/y2ksquad review?

**Pull request merged**